spinalcordtoolbox/scripts/sct_detect_compression.py

```diff
@@ -0,0 +1,316 @@
+        fname_seg = get_absolute_path(arguments.s)
+    if arguments.discfile is not None:
+        fname_disc = get_absolute_path(arguments.discfile)
+    torsion_slices = arguments.torsion_slices
+    angle_correction = bool(arguments.angle_corr)
+    param_centerline = ParamCenterline(
+        algo_fitting=arguments.centerline_algo,
+        smooth=arguments.centerline_smooth,
+        minmax=True)
+
+    # No SC segmentation provided, do it now
+    if arguments.s is None:
+        printv('Automatic segmentation is not implemented yet.', 1, 'error')
+        sys.exit(1)
+        # TODO - perform segmentation on fname_in within this script.
+
+    # No disc file provided, do it now
+    if arguments.discfile is None:
+        printv('Automatic labeling is not implemented yet.', 1, 'error')
+        sys.exit(1)
+        # TODO - perform disc identification (vertebral labeling) within this script.
+
+    # Compute morphometric metrics
+    metrics, fit_results = compute_shape(fname_seg,
+                                         angle_correction=angle_correction,
+                                         param_centerline=param_centerline,
+                                         verbose=verbose)
+
+    # Compute the average and standard deviation across slices
+    metrics_agg = {}
+    for key in ['area', 'diameter_AP', 'diameter_RL', 'solidity', 'orientation']:
+        # Note: we do not need to calculate all the metrics, we need just:
+        #   - area (will be CSA)
+        #   - diameter_AP and diameter_RL (used to calculate compression ratio)
+        #   - solidity
+        #   - orientation (used to calculate torsion)
+        # Note: we have to calculate metrics across all slices (perslice) to be able to compute orientation
+        metrics_agg[key] = aggregate_per_slice_or_level(metrics[key],
+                                                        perslice=True,
+                                                        perlevel=False, fname_vert_level=fname_disc,
+                                                        group_funcs=group_funcs
+                                                        )
+
+    metrics_agg_merged = merge_dict(metrics_agg)
+
+    # Compute compression ratio (CR) as 'diameter_AP' / 'diameter_RL'
+    # TODO - compression ratio (CR) could be computed directly within the compute_shape function -> consider that
+    for key in metrics_agg_merged.keys():           # Loop across slices
+        # Ignore slices which have diameter_AP or diameter_RL equal to None (e.g., due to bad SC segmentation)
+        if metrics_agg_merged[key]['MEAN(diameter_AP)'] is None or metrics_agg_merged[key]['MEAN(diameter_RL)'] is None:
+            metrics_agg_merged[key]['CompressionRatio'] = None
+        else:
+            metrics_agg_merged[key]['CompressionRatio'] = metrics_agg_merged[key]['MEAN(diameter_AP)'] / \
+                                                          metrics_agg_merged[key]['MEAN(diameter_RL)']
```

Comment:
Thank you for taking the initiative on this project!

On first glance, it seems to me that the new functionality is basically computing 3 new metrics:

* `CompressionRatio`
* `Torsion`
* Compression probability

My question, then, is: Would it be feasible to instead add these 3 metrics to the actual `compute_shape` function? That way, these 3 metrics will be included in the CSV file output by `sct_process_segmentation`.

Then, for the other parts of the pipeline (spinal cord segmenation, labeling, etc.), this could be demonstrated via e.g. a tutorial.

Reply:
Thank you for the feedback, @joshuacwnewton! I wrote the response several weeks ago, but apparently, I did not submit it. Sorry for that.
We have been now working on compression detection within the `normalization project`; see [description](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3733) and [PR](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3977). 
Since the code proposed in this PR works on a specific type of [data](https://pubmed.ncbi.nlm.nih.gov/35371944/), I would not merge this PR to master. But, please, do not delete the branch.
The more general compression detection (i.e., working across different data) will come within the `normalization project`.

Regarding `sct_process_segmentation` and the implementation of other metrics. Yeah, `CompressionRatio` would be useful. (`Torsion` is computed from several slices --> probably more difficult to implement. `Compression probability` is specific for this code --> no need to implement)

Reply:
Ahhh, I see! So, this branch/PR is more for experimentation purposes, rather than intended to be merged into SCT? 

Thank you for the clarification, and my apologies for the premature review!

Reply:
> So, this branch/PR is more for experimentation purposes, rather than intended to be merged into SCT?

Exactly.